pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/StagePlanner.java

```diff
@@ -84,6 +89,9 @@ public QueryPlan makePlan(RelRoot relRoot) {
       _workerManager.assignWorkerToStage(e.getKey(), e.getValue(), _requestId);
     }
 
+    // Run physical optimizations
+    runPhysicalOptimizers(queryPlan);
+
```

Comment:
can we change this to `runWorkerAssignmentStrategy(queryPlan, _workerManager) and get rid of the 4 lines above ?